src/menus/autoConnectScreen.cpp

```diff
@@ -65,36 +65,47 @@ void AutoConnectScreen::update(float delta)
     {
         std::vector<ServerScanner::ServerInfo> serverList = scanner->getServerList();
         string autoconnect_address = PreferencesManager::get("autoconnect_address", "");
+        int autoconnect_port = PreferencesManager::get("autoconnect_port").toInt();
```

Comment:
would make more sense to parse the address in `host:port` format, which is the usual way on how to specify this, and I'm quite sure that's also how the manual IP entry of connecting works.